#goal definition
char[] messageType
char[] messageContent
---
#result definition
bool isOK
---
#feedback
int32[] sequence
